### PR TITLE
Added a feature for updating the group names

### LINKED ISSUE #3451: Add an option to change user group name
### Is your proposal related to a problem?

Currently, We don't have a feature for changing the name of an existed user group

### Describe the solution you'd like

Add an option to change the user group's name 

### Describe alternatives you've considered

No

### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by shah21**
Resolves #3451

### CODE REVIEW

File: server/src/services/group_permissions.service.ts [hunk 1]
```diff
@@ -136,6 +137,31 @@ export class GroupPermissionsService {
...
 
     await getManager().transaction(async (manager) => {
+      //update user group name
+      if (name) {
+        const reservedGroups = ['admin', 'all_users'];
+        if (reservedGroups.includes(groupPermission.group)) {
+          throw new BadRequestException('Cannot update a defualt group name');
+        }
+
+        if (['All Users', 'Admin'].includes(name)) {
```
Comment (akshaysasidrn) [inline server/src/services/group_permissions.service.ts:147]: These are just displayed text for group defaults. Converting the name to snake_case and checking against reserved names should do.

**Review by akshaysasidrn — CHANGES_REQUESTED** [review]
Please add spec as well

**Comment by Nandinisaha13** [comment]
<html>
<body>
<google-sheets-html-origin>Test Cases

Test Cases | Steps | Result
-- | -- | --
Should be able to update group name. | Hover over workspace. | Options are displayed.
  | Click on manage groups. | Manage group is displayed.
  | Click on create new group button. | Creation of group page is displayed.
  | Enter the name of group & click on save. | The group is entered and it is created.
  | Click on Update button for the new group. | Editing page of group is displayed.
  | Enter updated name. | The name is enetered.
  | Click on save. | The group name is updated.
Should be able to cancel the editing of group name. | Hover over workspace. | Options are displayed.
  | Click on manage groups. | Manage group is displayed.
  | Click on create new group button. | Creation of group page is displayed.
  | Enter the name of group & click on save. | The group is entered and it is created.
  | Click on Update button for the new group. | Editing page of group is displayed.
  | Enter updated name. | The name is enetered.
  | Click on cancel. | The editing gets cancelled.
Should not be able to change the group name to previous group name | Hover over workspace. | Options are displayed.
  | Click on manage groups. | Manage group is displayed.
  | Click on create new group button. | Creation of group page is displayed.
  | Enter the name of group & click on save. | The group is entered and it is created.
  | Click on Update button for the new group. | Editing page of group is displayed.
  | Enter the previous name. | Save button is disabled.
Should not be able to edit to already existing group name. | Hover over workspace. | Options are displayed.
  | Click on manage groups. | Manage group is displayed.
  | Click on create new group button. | Creation of group page is displayed.
  | Enter the name of group & click on save. | The group is entered and it is created.
  | Click on create new group button. | Creation of group page is displayed.
  | Enter the name of group & click on save. | The group is entered and it is created.
  | Click on Update button for the new group. | Editing page of group is displayed.
  | Enter name as the existing group name. | The name is entered.
  | Click on save. | It throws toast message as group name exists


</body>
</html>

**Review by akshaysasidrn — DISMISSED** [review]

File: server/src/services/group_permissions.service.ts [hunk 2]
```diff
@@ -139,6 +140,31 @@ export class GroupPermissionsService {
     } = body;
 
     await getManager().transaction(async (manager) => {
+      //update user group name
+      if (name) {
+        const reservedGroups = ['admin', 'all_users'];
+        if (reservedGroups.includes(groupPermission.group)) {
+          throw new BadRequestException('Cannot update a defualt group name');
```
Comment (akshaysasidrn) [inline server/src/services/group_permissions.service.ts:147]: default*

**Comment by shubh22** [comment]
Task linked: [CU-2hcxvka Ability to change the user group name](https://app.clickup.com/t/2hcxvka)

**Comment by gsmithun4** [comment]
Screenshot 2022-07-06 at 7 55 24 PM
@shah21 create and update accepting empty spaces, please add validations in dto

**Review by gsmithun4 — CHANGES_REQUESTED** [review]
add validations in dto for group name

File: server/src/services/group_permissions.service.ts [hunk 3]
```diff
@@ -139,6 +140,35 @@ export class GroupPermissionsService {
...
+        }
+
+        const groupToFind = await this.groupPermissionsRepository.findOne({
+          where: {
+            organizationId: user.organizationId,
+            group: name,
+          },
+        });
+
+        if (groupToFind && groupToFind.id !== groupPermission.id) {
```
Comment (gsmithun4) [inline server/src/services/group_permissions.service.ts:165]: if `groupToFind.id !== groupPermission.id` we can skip updation right?

Comment (shah21) [reply server/src/services/group_permissions.service.ts:165]: Yes

**Review by akshaysasidrn — APPROVED** [review]

**Review by gsmithun4 — APPROVED** [review]
LGTM

**Review by gsmithun4 — APPROVED** [review]

**Pull request merged**